File: 0069-sqrtx/0069-sqrtx.go
func mySqrt(x int) int {

    // agar x 0 ya 1 hai
    if x<=1{
        return x
    }

    l:=1
    h:=x
    ans:=0

    // binary search chalao
    for l<=h{

        m:=l+(h-l)/2

        // agar square root mil gaya
        if m==x/m{
            return m
        }

        // agar square chota hai
        if m<x/m{

            ans=m
            l=m+1

        }else{

            // warna left mei jao
            h=m-1
        }
    }

    // answer return kardo
    return ans
}

File: 0069-sqrtx/0069-sqrtx.cpp
class Solution {
public:

    int mySqrt(int x){

        // agar x 0 ya 1 hai
        if(x<=1){
            return x;
        }

        long long l=1,h=x;
        int ans=0;

        // binary search chalao
        while(l<=h){

            long long m=l+(h-l)/2;

            // agar square equal hai
            if(m*m==x){
                return m;
            }

            // agar square chota hai
            if(m*m<x){

                ans=m;
                l=m+1;
            }

            // warna left mei jao
            else{
                h=m-1;
            }
        }

        // answer return kardo
        return ans;
    }
};